11_rtc_driver/ds3231x_drv.c

```diff
@@ -0,0 +1,194 @@
+﻿// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/i2c.h>
+
+#include <linux/init.h>
+#include <linux/rtc.h>
+#include <linux/bcd.h>
+#include <linux/regmap.h>
+
+#include "ds3231_regs.h"
+
+MODULE_AUTHOR("Andrii Synenko");
+MODULE_DESCRIPTION("ds3231 driver for Linux Kernel ProCamp");
+MODULE_LICENSE("Dual BSD/GPL");
+
+static const struct i2c_device_id ds3231x_id[] = {
+	{"ds3231x"}, {}
+};
+MODULE_DEVICE_TABLE(i2c, ds3231x_id);
+static const struct of_device_id ds3231x_of_match[] = {
+	{.compatible = "maxim,ds3231x" }, {}
```

Comment:
Keep `{}` on separate line.
This minimize diff after adding a new compatible string (no lines is changed, just one is added)

Reply:
Fixed